OmsiHook/OmsiComplMapObj.cs

```diff
@@ -223,5 +224,22 @@ public OmsiCameraSettings[] ReflCameraSettings
             get => new(Memory, Address + 0x260);
         }
 
+
+        public int GetVarIndex(string VarName)
+        {
+            var strings = this.VarStrings;
+            for (int i = 0; i < strings.arrayCache.Length; i++)
```

Comment:
Make `MemArrayBase.arrayCache` private; no one should be reading from it directly always use `WrappedArray`.
Also, `MemArray`s won't cache as you expect unless you make them singletons inside the `OmsiComplMapObj` class. At the moment every time you call this it will construct a new MemArray, read the whole array from game memory and then perform a linear search through the cached array. This is incredibly slow and puts a lot of stress on the GC. Ideally, we should have a dictionary-backed cache as well for faster lookups; this could be implemented as a decorator class for `MemArrayString` and would expose a `Dictionary<string, int> WrappedIndexMap { get; }`.

Reply:
The singleton issue applies to all uses of `MemArray` might need to think how best to approach it.